authz/sdk_end2end_test.go

```diff
@@ -379,6 +379,30 @@ func (s) TestSDKFileWatcherEnd2End(t *testing.T) {
 	}
 }
 
+func verifyValidReload(ctx context.Context, tsc pb.TestServiceClient, wantCode codes.Code, wantErr string) (lastStatus *status.Status) {
+	for numRetries := 0; numRetries <= 20; numRetries++ {
+		_, err := tsc.UnaryCall(ctx, &pb.SimpleRequest{})
+		if lastStatus = status.Convert(err); lastStatus.Code() == wantCode && lastStatus.Message() == wantErr {
+			return nil
+		}
+		time.Sleep(20 * time.Millisecond)
+		numRetries++
+	}
+	return lastStatus
```

Comment:
You didn't like my version which returned `error`?  Returning an `error` (nil on success, non-nil on error) is more natural for go.

Reply:
Optionally, instead of returning the RPC error itself, return `fmt.Errorf("got %v; want %v", lastStatus.Err(), wantStatus.Err())` and at the caller, do simply `if err != nil { t.Fatal("<optional extra info>", err) }`

Reply:
Oh I had done that to avoid converting error to status for failure log. Since now we would be logging error, reverted the function to error. Let me know if this looks good.